test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -23,29 +22,62 @@ import 'boilerplate_utilities_test.dart';
 
 void main() {
   group('AdvancedPropsAndStateClassMigrator', () {
-    final converter = ClassToMixinConverter();
-    final testSuggestor =
-        getSuggestorTester(AdvancedPropsAndStateClassMigrator(converter));
+    group('', () {
+      AdvancedPropsAndStateClassMigratorTestHelper(
+        path: 'test/boilerplate_suggestors/semver_report.json',
+        shouldTreatAllComponentsAsPrivate: false,
+        isValidFilePath: true,
+      );
+    });
 
-    setUpAll(() {
-      semverHelper = SemverHelper(jsonDecode(reportJson));
+    group('with --treat-all-components-as-private flag', () {
+      AdvancedPropsAndStateClassMigratorTestHelper(
+        path: 'test/boilerplate_suggestors/semver_report.json',
+        shouldTreatAllComponentsAsPrivate: true,
+        isValidFilePath: true,
+      );
     });
 
-    tearDown(() {
-      converter.setConvertedClassNames({});
+    group('with invalid file path', () {
+      AdvancedPropsAndStateClassMigratorTestHelper(
+        path: 'test/boilerplate_suggestors/does_not_exist.json',
+        shouldTreatAllComponentsAsPrivate: false,
+        isValidFilePath: false,
+      );
     });
+  });
+}
 
-    group('does not operate when', () {
-      test('it\'s an empty file', () {
-        testSuggestor(expectedPatchCount: 0, input: '');
+void AdvancedPropsAndStateClassMigratorTestHelper({
+  String path,
+  bool shouldTreatAllComponentsAsPrivate,
+  bool isValidFilePath,
+}) {
+  final converter = ClassToMixinConverter();
+  final semverHelper = getSemverHelper(path,
+      shouldTreatAllComponentsAsPrivate: shouldTreatAllComponentsAsPrivate);
+  SuggestorTester testSuggestor;
+
+  setUpAll(() async {
+    testSuggestor = getSuggestorTester(
+        AdvancedPropsAndStateClassMigrator(converter, await semverHelper));
```

Comment:
Was there a reason the `await` for `semverHelper` is on this line rather than line 57? I usually see `await`s on the line of the instantiation, rather than the usage.